File: src/Queryable/Abstractions/Enumerations.cs
using System.ComponentModel;
using System.Text.Json.Serialization;

namespace Kaleido.Queryable;
/// <summary>Defines the filter operations supported by the framework query model.</summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum FilterOperator
{
    // Equality
    [Description("Equals")]
    Equals,
    [Description("Not Equals")]
    NotEquals,

    // Comparison
    [Description("Greater Than")]
    GreaterThan,
    [Description("Less Than")]
    LessThan,
    [Description("Greater Than Or Equal")]
    GreaterThanOrEqual,
    [Description("Less Than Or Equal")] 
    LessThanOrEqual,

    // String
    [Description("Contains")]
    Contains,
    [Description("Not Contains")]
    NotContains,
    [Description("Starts With")]
    StartsWith,
    [Description("Ends With")]
    EndsWith,

    // Set
    [Description("In")]
    In,
    [Description("Not In")]
    NotIn,

    [Description("Between")]
    Between,
    [Description("Not Between")]
    NotBetween,

    // Null
    [Description("Is Null")]
    IsNull,
    [Description("Is Not Null")]
    IsNotNull,

    // Boolean
    [Description("Is True")]
    IsTrue,
    [Description("Is False")]
    IsFalse,

    //// Collection
    //Any,
    //All,

    //// Advanced
    //Regex,
    //Like,
    //NotLike
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum MatchMode
{
    [Description("Exact Match")]
    Exact,
    [Description("Starts With")]
    StartsWith,
    [Description("Ends With")]
    EndsWith,
    [Description("Contains")]
    Contains
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum SortDirection
{
    [Description("Ascending")]
    Ascending,
    [Description("Descending")]
    Descending
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum LogicalOperator
{
    [Description("And")]
    And,
    [Description("Or")]
    Or
}

public enum ValueDataType 
{ 
    String, 
    Integer,
    Long,
    Decimal, 
    Boolean, 
    Guid,
    Date,
    DateTime,
    Time,
    Enum, 
    Unknown
}

File: samples/kaleido-sample-ecommerce-ui/src/app/kaleido/models/enumerations.ts
export enum FilterOperator {
  // Equality
  Equals = 'Equals',
  NotEquals = 'NotEquals',
  // Comparison
  GreaterThan = 'GreaterThan',
  LessThan = 'LessThan',
  GreaterThanOrEqual = 'GreaterThanOrEqual',
  LessThanOrEqual = 'LessThanOrEqual',
  // String
  Contains = 'Contains',
  NotContains = 'NotContains',
  StartsWith = 'StartsWith',
  EndsWith = 'EndsWith',
  // Set
  In = 'In',
  NotIn = 'NotIn',
  Between = 'Between',
  NotBetween = 'NotBetween',
  // Null
  IsNull = 'IsNull',
  IsNotNull = 'IsNotNull',
  // Boolean
  IsTrue = 'IsTrue',
  IsFalse = 'IsFalse'
}
export enum MatchMode {
  Exact = 'Exact',
  StartsWith = 'StartsWith',
  EndsWith = 'EndsWith',
  Contains = 'Contains'
}
export enum SortDirection {
  Ascending = 'Ascending',
  Descending = 'Descending'
}
export enum LogicalOperator {
  And = 'And',
  Or = 'Or'
}

export const FILTER_OPERATOR_LABELS:
  Record<FilterOperator, string> =
{
  [FilterOperator.Equals]:
    'Equals',

  [FilterOperator.NotEquals]:
    'Not Equals',

  [FilterOperator.GreaterThan]:
    'Greater Than',

  [FilterOperator.LessThan]:
    'Less Than',

  [FilterOperator.GreaterThanOrEqual]:
    'Greater Than Or Equal',

  [FilterOperator.LessThanOrEqual]:
    'Less Than Or Equal',

  [FilterOperator.Contains]:
    'Contains',

  [FilterOperator.NotContains]:
    'Not Contains',

  [FilterOperator.StartsWith]:
    'Starts With',

  [FilterOperator.EndsWith]:
    'Ends With',

  [FilterOperator.In]:
    'In',

  [FilterOperator.NotIn]:
    'Not In',

  [FilterOperator.Between]:
    'Between',

  [FilterOperator.NotBetween]:
    'Not Between',

  [FilterOperator.IsNull]:
    'Is Null',

  [FilterOperator.IsNotNull]:
    'Is Not Null',

  [FilterOperator.IsTrue]:
    'Is True',

  [FilterOperator.IsFalse]:
    'Is False'
};

export const MATCH_MODE_LABELS:
  Record<MatchMode, string> =
{
  [MatchMode.Exact]:
    'Exact Match',

  [MatchMode.StartsWith]:
    'Starts With',

  [MatchMode.EndsWith]:
    'Ends With',

  [MatchMode.Contains]:
    'Contains'
};